IngFerchos/ct-candidates-app
# Add Task List Features

Overview:
This pull request implements a Task List application with user authentication and task management features. Below are the details of the changes made:

Features Implemented:

Task Management:

Task Creation: Users can create new tasks with a title, order, and completion status.
Task Viewing: Users can view a list of all tasks.
Task Editing: Users can edit the details of existing tasks.
Task Deletion: Users can delete tasks from the list.
Filtering and Sorting:

Task Filtering: Users can filter tasks by title.
Task Sorting: Users can sort tasks by title, order, and completion status.
User Authentication:

User Registration: Implemented using Laravel Breeze. After registration, users are redirected to the login page instead of being logged in automatically.
User Login: Users can log in to the application.
User Association: Tasks are associated with the authenticated user. Each user can only view and manage their own tasks.
Styling and User Interface:

Bootstrap Integration: Added Bootstrap for responsive and modern UI design.
Font Awesome Icons: Used Font Awesome for sorting icons to improve user experience.
Details of Implementation:

Backend:

Controllers:
TaskController handles all CRUD operations for tasks.
Modified RegisteredUserController to redirect to the login page after registration.
Models:
Updated the Task model to include a relationship with the User model.
Migrations:
Added a migration to include user_id in the tasks table.

Frontend:

Views:
Created views for task management (index, create, edit, show).
Updated the login view to include a link to the registration page.
Components:
Used Bootstrap components for a modern and responsive design.
Integrated Font Awesome for sorting icons.
Testing:

Verified the functionality of task creation, viewing, editing, and deletion.
Tested user registration and login flows.
Confirmed task filtering and sorting work as expected.
How to Test:

Clone the repository and check out the branch feature/hugo-cabezas-task-list.
Run composer install to install dependencies.
Run php artisan migrate to apply migrations.
Run npm install && npm run dev to compile assets.
Run php artisan serve to start the development server.
Navigate to http://localhost:8000 and test the application.
Screenshots:
![image](https://github.com/user-attachments/assets/927df5c9-bb46-4db5-9d0d-0bc16d53fa3d)
![image](https://github.com/user-attachments/assets/75b3ad81-c303-474a-86ef-7ef081e2795e)
![image](https://github.com/user-attachments/assets/7793b2c6-1da0-4b0d-b4a3-c771a0d002c1)
![image](https://github.com/user-attachments/assets/e5973615-813f-41e8-86b0-109f6024ae29)
![image](https://github.com/user-attachments/assets/7f8c3cac-a28f-42fd-95f0-82dd0c4a86fc)
